### ISSUE TITLE
GPU Extension not detected in 2022

### ISSUE THREAD

**Issue by AddioElectronics** [dotnet/machinelearning-modelbuilder#2247]
While trying to create a MLModel, I selected Local (GPU), checked the compatibility and installed all the required components.
The only thing that is not working properly is the GPU Extension, which says "GPU extension is missing. Install extension."
When I click "Install Extension" it directs me to the VS 2019 extension.
I manually installed the 2022 extension from the Extension Manager, and it is not being detected.
I also tried installing the 2019 version with no luck.

My graphics card is a GTX 1080.

One thing I noticed is that installing CUDA10.1 installs Nsight for VS2019, and not 2022. If I attempt to install for VS 2022, it wants to uninstall the older version, which I don't believe has the required Cuda version of 10.1.
Is VS 2019 supposed to have an option to select a Local GPU for the environment? Because it does not exist for me.

**Comment by beccamc** [comment]
Sorry you've run into this problem. The Model Builder and GPU version **need to match**. I assume you are on an older version of Model Builder because in newer version we have updated the error message to indicate that. 

Can you update both [Model Builder](https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilder2022) and [GPU](https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilderGPU2022) to 16.13.10.2241902?

**Comment by beccamc** [comment]
Any luck with updating the versions? Did that solve the problem?

**Comment by ahazelwood** [comment]
I am having the same issue (GeForce RTX 3050).  Version of the Model Builder and GPU are the version mentioned above (16.13.10.2241902).  I do not get the option to train on the GPU, only on the CPU.  Windows 11, Visual Studio 2022 (64-bit) Version 17.3.4

**Comment by beccamc** [comment]
We only support GPU for Image Classification. Are you trying to run GPU for a different scenario?

**Comment by ahazelwood** [comment]
Yes.  I was trying to use it for Text Classification.

**Comment by beccamc** [comment]
Sorry, text classification isn't supported. You can checkout our [FAQ](https://github.com/dotnet/machinelearning-modelbuilder/blob/main/troubleshooting/ModelBuilderFAQ.md#why-isnt-gpu-supported-for-my-scenario) for links to more details.

**Comment by beccamc** [comment]
Closing this issue. If GPU doesn't show up for you one of these maybe be the problem... 

1. Model Builder extension and Model Builder GPU extension need to be the same version number
2. Image Classification is the only supported scenario. See this [FAQ](https://github.com/dotnet/machinelearning-modelbuilder/blob/main/troubleshooting/ModelBuilderFAQ.md#why-isnt-gpu-supported-for-my-scenario) for more details 

Feel free to comment if the above doesn't work.

**Comment by AddioElectronics** [comment]
Sorry I forgot to come check for replies. I will try updating thanks.

**Comment by AddioElectronics** [comment]
Turns out I needed to update the Model Builder Extension, the link from the model builder was not only to the wrong 2019 version, but also the wrong extension in general.

When I freshly installed VS 2022, the Visual Studio Installer must have installed an older version of model builder.

Installing both of these worked for me.
https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilder2022
https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilderGPU2022
